share/share.go

```diff
@@ -18,15 +18,18 @@ var (
 )
 
 const (
-	// MaxSquareSize is currently the maximum size supported for unerasured data in
-	// rsmt2d.ExtendedDataSquare.
-	MaxSquareSize = appconsts.MaxSquareSize
 	// NamespaceSize is a system-wide size for NMT namespaces.
 	NamespaceSize = appconsts.NamespaceSize
 	// Size is a system-wide size of a share, including both data and namespace ID
 	Size = appconsts.ShareSize
 )
 
+var (
+	// MaxSquareSize is currently the maximum size supported for unerasured data in
+	// rsmt2d.ExtendedDataSquare.
+	MaxSquareSize = appconsts.SquareSizeUpperBound(appconsts.LatestVersion)
```

Comment:
I wonder if we can just set this to ipld.MaxSquareSize

Reply:
https://github.com/celestiaorg/celestia-node/pull/2305/commits/4719e5130a6e685048470cd140cba0d2765f27b2